### ISSUE TITLE
REST API Calls (GET) fail when "Bigint" types are passed in query

### ISSUE THREAD

**Issue by adas98012** [hasura/graphql-engine#8397]
### Version Information

Server Version: 2.3.1

### Environment

Cloud

### What is the expected behaviour?

RESTified endpoints should support and correctly parse bigint datatypes in query string.

### Keywords

Bigint, REST API

### What is the current behaviour?

When a bigint argument is part of the query string, it will be parsed as a number instead of a bigint, which results in a type error:  **Expected bigint type for variable id  got Number**

### How to reproduce the issue?

1. Create a hasura cloud project and add a database.
2. Enable stringified numeric types by setting the following environment variable: `HASURA_GRAPHQL_STRINGIFY_NUMERIC_TYPES` to `true`
3. Create and track table with a bigint column:
```sql
CREATE TABLE bigint_test (
  id bigint NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
  name TEXT
);
```
4. Create a [RESTified Graphql endpoint](https://hasura.io/docs/latest/graphql/core/api-reference/restified/) using this table with a bigint argument, for example using this query:
```graphql
query MyQuery_BigInt($id: bigint!) {
  bigint_test (where: { id: { _eq: $id } }) {
    id
    name
  }
}
```
5. Configure the endpoint to accept `GET` requests and test it, passing the id as a variable.

![Screen Shot 2022-04-13 at 1 36 05 PM](https://user-images.githubusercontent.com/101733366/163266260-4c813768-fc77-4f50-a477-09892efac5d2.png)
6. An error is returned:  **Expected bigint type for variable id  got Number**

**Seen on Hasura Console:** 

![Screen Shot 2022-04-13 at 1 43 38 PM](https://user-images.githubusercontent.com/101733366/163267305-fae15b32-e200-4351-8b35-099cab1ca0c9.png)

**Reported by User from their Log:**

![Screen Shot 2022-04-13 at 1 46 59 PM](https://user-images.githubusercontent.com/101733366/163267730-d015f791-7151-4d82-afbf-9fc4b355e862.png)
### Any possible solutions?

Bigint arguments are parsed correctly if passed as part of the request body, instead of being part of the query string.

**Comment by adimyth** [comment]
I'm facing the same issue. @adas98012 did u find any solution?

**Comment by SkySor44** [comment]
I'm also facing this issue, any help would be appreciated

**Comment by adas98012** [comment]
> I'm also facing this issue, any help would be appreciated
You can use the REST API as a POST (instead of GET) and pass the "id" in the body of the request.

**Comment by hallettj** [comment]
Hi @adas98012; I believe that this problem is fixed in v2.7.0 which is currently in beta pre-release. I followed the reproduction steps that you (very helpfully) supplied. I got the same error when testing in v2.3.1, but the request seems to work correctly in v2.7.0-beta.1.

The relevant changelog entry is:

> server: makes url/query variables in REST endpoints assume string if other types not applicable
